spec/mongoid/matcher/extract_attribute_data/traversal.yml

```
@@ -257,3 +257,34 @@
   exists: false
   value: ~
   expanded: false
+
+- name: numeric Hash key inside expanded array
+  document: &numeric-hash-keys-in-expanded-array
+    a:
+      '1':
+        b:
+          -
+            c:
+              '2': 3
+          -
+            d:
+              '4': 5
+          - e: f
```

This could be formatted more similarly to the other keys and have the same semantic meaning

```suggestion
          -
            e: f
```